### ISSUE TITLE
Please add support for logging in with an account on TorrentGalaxy

### ISSUE THREAD

**Issue by NoX1De** [Prowlarr/Indexers#391]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

TorrentGalaxy supports logging in with a user account, but the indexer definition does not support this for some reason, being authenticated with an account on the website is beneficial for a number of reasons that I could explain if need be.

### Describe the solution you'd like

Please add a login block to https://github.com/Prowlarr/Indexers/blob/b7798267fab07d44b1d73a1e2a21ae552df78eaa/definitions/v7/torrentgalaxy.yml that allows configurable username and password for login in addition to the captcha settings.

### Describe alternatives you've considered

N/A

### Anything else?

Thank you !

**Comment by bakerboy448** [comment]
@ilike2burnthing @mynameisbogdan

**Comment by ilike2burnthing** [comment]
From a quick look, the only thing I could see as potentially being useful for a minority of users would be disabling adult content. Is there anything else I'm missing? Even signed in they're still just using public trackers.

There's also seemingly an API for those ranked verified uploaders or higher, but that would need a whole new indexer and an account with access to the API, and would ultimately benefit next to no one.

**Comment by ilike2burnthing** [comment]
Really? I haven't been having any major issues with it. The odd time I'll see a healthcheck warning for it in the *arrs, but it either resolves itself in the next RSS sync or I reconfigure the indexer with the new captcha.

**Comment by NoX1De** [comment]
Adding support for logging in appears to reduce captcha frequency in cursory manual testing...I'm hoping this makes the indexer more useful.

**Comment by NoX1De** [comment]
> Well, many complaints of it not working and considering there's a captcha doesn't make it very automation friendly. 🤷‍♂️

See my comment above, this is a top notch source from a content quality perspective, if logging in with an account via Prowlerr also greatly reduces the captcha (being logged in manually seems to mostly fully eliminate it, based on my testing today) as it does manually than this change would be a significant improvement.

**Comment by bakerboy448** [comment]
`TorrentGalaxy` and `this is a top notch source from a content quality perspective` is an oxymoron of a sentence

Shitty public tracker that inflates and lies about its seed count & is anti-automation given it's increased captcha is rather comical

but anyway. maybe someone with enough drive to implement the log in version will come around, one day... maybe

**Comment by ilike2burnthing** [comment]
We could have a second semi-private login indexer, but it'd need to be able to handle both the login and galaxy fence pages, which Cardigann can't currently do.

Either the second indexer would need to be written in C# or Cardigann adapted to:

1) fetch captcha from `galaxyfence.php`
2) submit user's captcha response
3) post login inputs to `account-login.php`
4) load test path and check for selector

**Comment by NoX1De** [comment]
> `TorrentGalaxy` and `this is a top notch source from a content quality perspective` is an oxymoron of a sentence
> 
> Shitty public tracker that inflates and lies about its seed count & is anti-automation given it's increased captcha is rather comical
> 
> but anyway. maybe someone with enough drive to implement the log in version will come around, one day... maybe

I agree they are a pain but I'm sure they have their own expenses they have to modulate/minimize, who knows their motivation for being such a pain but reducing operating expenses would be my first guess...agree to disagree on available content there, lots of quality 4K/UHD not found elsewhere in public tracker land, at least not that I'm currently aware of.  Never really cared to pay attention to any kind of supposed seed count fraud. lol.

> We could have a second semi-private login indexer, but it'd need to be able to handle both the login and galaxy fence pages, which Cardigann can't currently do.
> 
> Either the second indexer would need to be written in C# or Cardigann adapted to:
> 
> 1. fetch captcha from `galaxyfence.php`
> 2. submit user's captcha response
> 3. post login inputs to `account-login.php`
> 4. load test path and check for selector

This is along the lines of what I've been pondering was also thinking this should be semi-private and a second indexer definition for that makes a ton of sense, basically had been wondering if this can currently be done in a definition, sounds like that's a no.  

Any idea how much effort adapting Cardigann would be to enable this functionality?  Or which option would be the least effort, I'm not currently familiar with Cardigann's codebase or how it's being leveraged wrt to Prowlarr indexers.   At least it's written in golang...heh.  Just taking a super quick glance, it looks like the following file would need to be adapated, perhaps?

https://github.com/cardigann/cardigann/blob/4ad57c5e08e96f86bf08e5f97db40f458a29e119/indexer/parser.go

**Comment by ilike2burnthing** [comment]
`Prowlarr/{Prowlarr version} ({platform} {platform version})`
e.g.:
`Prowlarr/1.10.5.4116 (windows 10.0.19041.0)`

No, it doesn't trigger a captcha challenge (or at least not immediately).

**Comment by soredake** [comment]
> I'm pretty confident that being logged in makes a substantial difference in the frequency of being prompted for a captcha after testing this for a week or so.

For me, it changed nothing, I registered on tgx, after 3 days testing I got like 10 captchas, no difference.

**Comment by NoX1De** [comment]
> > I'm pretty confident that being logged in makes a substantial difference in the frequency of being prompted for a captcha after testing this for a week or so.
> 
> For me, it changed nothing (with normal browser or prowlarr user-agent), I registered on tgx, after 3 days testing I got like 10 captchas, no difference.

Hmmm well that's discouraging.  I never got around to testing this with the Prowlarr user-agent but when I did test with normal browser it seemed to reduce the CAPTCHA frequency significantly.  I wonder if they got even more aggressive with logged in accounts since then....shrug, really sucks too because I think they are one of the best in terms of content breadth, clearly not with implementing access policy for their site.

**Comment by mikekuzak** [comment]
> drive

Any public alternative? Rarbg style?

**Comment by ilike2burnthing** [comment]
TheRARBG

**Comment by Pengi123** [comment]
TorrentGalaxy remove "galaxy fence" so now there is no captchas, I hope Prowlarr will again add TorrentGalaxy in list.
